### PR TITLE
Parameter verification fixes

### PR DESCRIPTION
**Pull request by toml01**

### CODE REVIEW

File: cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs [hunk 1]
```diff
@@ -152,3 +151,113 @@ fn decode_multisig_signature(raw_blob: &[u8]) -> Result<MultisigSignature, Crypt
...
+            0, 0, 0, 0, 0, 0, 0, 0x12, 205, 1, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15,
+        ];
+
+        let result = decode_multisig_signature(malformed_sig.as_slice());
+        assert!(
+            result.is_err(),
+            "Signature is: {:?} and result is: {:?}",
+            malformed_sig,
+            result
+        );
```
Comment (reuvenpo) [inline cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs:222]: you can probably assert equality with a specific error, if the error type implements `PartialEq` and either `Debug` or `Display` (not sure which one). you can import a quick and dirty `Display` derive macro from `derive_more::Display` which is available

**Review by reuvenpo — COMMENTED** [review]
great fixes so far :muscle::smile:
Just left some minor aesthetic comments on the tests.

File: cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs [hunk 2]
```diff
@@ -152,3 +151,113 @@ fn decode_multisig_signature(raw_blob: &[u8]) -> Result<MultisigSignature, Crypt
 
     Ok(signatures)
 }
+
+#[cfg(feature = "test")]
+pub mod tests_decode_multisig_signature {
+    use crate::crypto::multisig::{decode_multisig_signature, MultisigSignature};
+
```
[the remark this answers is not in the document]
Comment (toml01) [reply cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs:158]: Yep, I renamed it so I could create another test module in the future, e.g. `pub mod tests_multisig_pubkey`

File: cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs (hunk 1 above)
Comment (toml01) [reply cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs:222]: Not sure that it makes a difference lol

Comment (toml01) [reply cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs:222]: besides, all errors there are `CryptoError::ParsingError` so..

Comment (reuvenpo) [reply cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs:223]: The main difference is in the error message if it fails, but yeah this isn't really important in this case.

File: cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (reuvenpo) [reply cosmwasm/packages/wasmi-runtime/src/crypto/multisig.rs:165]: fair enough i suppose.

File: x/compute/internal/keeper/keeper.go [hunk 3]
```diff
@@ -124,37 +124,56 @@ func GetSignerSignature(signer exported.Account, tx auth.StdTx) (authtypes.StdSi
...
+	// https://github.com/cosmos/cosmos-sdk/commit/c13809062ab16bf193ad3919c77ec03c79b76cc8#diff-a64b9f4b7565560002e3ac4a5eac008bR148
+	tx := authtypes.StdTx{}
+	txBytes := ctx.TxBytes()
+	err := k.cdc.UnmarshalBinaryLengthPrefixed(txBytes, &tx)
+	if err != nil {
+		return authtypes.StdSignature{
+			PubKey:    secp256k1.PubKeySecp256k1{},
+			Signature: []byte{},
+		}, nil, sdkerrors.Wrap(types.ErrInstantiateFailed, fmt.Sprintf("Unable to decode transaction from bytes: %s", err.Error()))
+	}
```
Comment (reuvenpo) [inline x/compute/internal/keeper/keeper.go:138]: To make these slightly nicer, you can assign the empty `authtypes.StdSignature` to a local in the beginning of the function, and just refer to it in the returns.

**Review by reuvenpo — APPROVED** [review]
:100:

**Pull request merged**